ngi_reports/ngi_reports.py

```diff
@@ -10,39 +10,92 @@
+allowed_report_types = [
+    fl.replace(".md", "")
+    for fl in os.listdir(
+        os.path.realpath(
+            os.path.join(
+                os.path.dirname(__file__), os.pardir, "data", "report_templates"
+            )
+        )
+    )
+] + ["ign_aggregate_report"]
 
-def make_reports (report_type, working_dir=os.getcwd(), config_file=None, **kwargs):
+
+def proceed_or_not(question):
+    yes = set(["yes", "y", "ye", "Y", "YES", "YE", "Yes"])
+    no = set(["no", "n", "N", "NO", "No"])
+    sys.stdout.write("{}".format(question))
+    while True:
+        choice = input().lower()
+        if choice in yes:
+            return True
+        elif choice in no:
+            return False
+        else:
+            sys.stdout.write("Please respond with 'yes' or 'no' ")
+
+
+def make_reports(report_type, working_dir=os.getcwd(), config_file=None, **kwargs):
 
     # Setup
-    template_fn = '{}.md'.format(report_type)
-    LOG.info('Report type: {}'.format(report_type))
+    template_fn = "{}.md".format(report_type)
+    LOG.info("Report type: {}".format(report_type))
 
     # use default config or override it if file is specified
     config = report_config.load_config(config_file)
 
     # Import the modules for this report type
-    report_mod = __import__('ngi_reports.reports.{}'.format(report_type), fromlist=['ngi_reports.reports'])
+    report_mod = __import__(
+        "ngi_reports.reports.{}".format(report_type), fromlist=["ngi_reports.reports"]
+    )
 
     proj = Project()
-    proj.populate(LOG, config._sections['organism_names'], **kwargs)
+    proj.populate(LOG, config._sections["organism_names"], **kwargs)
 
     # Make the report object
     report = report_mod.Report(LOG, working_dir, **kwargs)
 
     # Work out all of the directory names
     output_dir = os.path.realpath(os.path.join(working_dir, report.report_dir))
-    reports_dir = os.path.realpath(os.path.join(os.path.dirname(__file__), os.pardir, 'data', 'report_templates'))
+    reports_dir = os.path.realpath(
+        os.path.join(os.path.dirname(__file__), os.pardir, "data", "report_templates")
+    )
+    working_base_dir = os.path.split(os.getcwd())[1]
+
+    # check if the current dir is correct
+    if not working_base_dir == report.project:
+        question = "The current directory {} does not belong to the chosen project {}. Continue? ".format(
+            working_dir, report.project
+        )
+        if proceed_or_not(question):
+            LOG.info(
+                "The reports for project {} will be generated in the current directory {}".format(
+                    report.project, working_dir
+                )
+            )
+        else:
+            LOG.error(
+                "Exiting the script as the current directory does not belong to the chosen project"
+            )
+            return
```

Comment:
Technically this `return` statement will cause the script to exit successfully with exit code 0. That might fool automation tools such as Nextflow or pipes. Not sure what the recommendation is, either raise an exception or use sys.exit with a non-zero code.

Reply:
again, I did copy the logic from previous code, which of course can be changed, but then we should probably change it there too.

Reply:
If the code you copied is not in `ngi_reports`, you could change it in both places, otherwise change it here and make an issue on the repo its copied from.